tests/integration-tests/cfn_stacks_factory.py

```diff
@@ -307,6 +307,21 @@ def delete_all_stacks(self):
                     )
                 )
 
+    def delete_stacks_retain_ad(self):
```

Comment:
I suggest to not have a separate function. Let us imagine if tomorrow we should do the same for N other auxiliary stacks to be retained: with this approach we would have N new functions.

I suggest to extend the logic of the original delete_all_stacks by adding an optional parameter, such as "excluded_stacks" (a list).